relay-filter/src/config.rs

```diff
@@ -192,6 +196,102 @@ impl LegacyBrowsersFilterConfig {
     }
 }
 
+/// Configuration for a generic filter.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GenericFilterConfig {
+    /// Specifies whether this filter is enabled.
+    pub is_enabled: bool,
+    /// The condition for the filter.
+    pub condition: Option<RuleCondition>,
+}
+
+impl GenericFilterConfig {
+    /// Returns true if the filter is not enabled or no condition was supplied.
+    pub fn is_empty(&self) -> bool {
+        !self.is_enabled || self.condition.is_none()
+    }
+}
+
+/// Configuration for a set of ordered filters.
+#[derive(Clone, Debug, Default)]
+pub struct OrderedFilters(pub Vec<(String, GenericFilterConfig)>);
+
+impl OrderedFilters {
+    /// Returns true if there are no generic filters configured.
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty() || self.0.iter().all(|(_, value)| value.is_empty())
+    }
+}
+
+impl Serialize for OrderedFilters {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(self.0.len()))?;
+        for (filter_name, filter_config) in self.0.iter() {
+            map.serialize_entry(filter_name, filter_config)?
+        }
+        map.end()
+    }
+}
+
+struct OrderedFiltersVisitor();
+
+impl<'de> Visitor<'de> for OrderedFiltersVisitor {
+    type Value = OrderedFilters;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter
+            .write_str("a map where keys are filter names and values are filter configurations")
+    }
+
+    fn visit_map<M>(self, mut access: M) -> Result<Self::Value, M::Error>
+    where
+        M: MapAccess<'de>,
+    {
+        let mut filters = OrderedFilters(vec![]);
+
+        // We don't perform any kind of de-duplication in case of duplicate keys, since this might
+        // lead to an opaque behavior from the outside. The resolution of multiple filters with
+        // the same name, will be a responsibility of the matching algorithm.
```

Comment:
> The resolution of multiple filters with the same name, will be a responsibility of the matching algorithm.

Sending multiple filters with the same name should be sentry's responsibility, and not Relays to know if two filters are the same. How should Relay deduplicate two different filters with the same name? Relay should just apply them.

Reply:
This comment entails that Relay will naively match first in first out independently of duplicates, exactly because it's not Relay's responsibility.